pyoxidizer/src/py_packaging/distribution.rs

```diff
@@ -360,24 +475,170 @@ impl ParsedPythonDistribution {
     }
 
     /// Ensure pip is available to run in the distribution.
-    pub fn ensure_pip(&self) -> PathBuf {
-        let pip_path = self
-            .python_exe
-            .parent()
-            .expect("could not derive parent")
-            .to_path_buf()
-            .join(PIP_EXE_BASENAME);
+    pub fn ensure_pip(&self, logger: &slog::Logger) -> PathBuf {
+        let dist_prefix = self.base_dir.join("python").join("install");
+        let python_paths = resolve_python_paths(&dist_prefix, &self.version);
+
+        let pip_path = python_paths.bin_dir.join(PIP_EXE_BASENAME);
 
         if !pip_path.exists() {
-            std::process::Command::new(&self.python_exe)
-                .args(&["-m", "ensurepip"])
-                .status()
-                .expect("failed to run ensurepip");
+            warn!(logger, "{} doesnt exist", pip_path.display().to_string());
+            invoke_python(&python_paths, &logger, &["-m", "ensurepip"]);
         }
 
         pip_path
     }
 
+    /// Duplicate the python distribution, with distutils hacked
+    pub fn create_hacked_base(&self, logger: &slog::Logger) -> PythonPaths {
+        let venv_base = self.venv_base.clone();
+
+        let venv_dir_s = self.venv_base.display().to_string();
+
+        if !venv_base.exists() {
+            if self.os == "windows" {
+                let external_python = which("python").unwrap().clone();
```

Comment:
this should be done much earlier, so the user doesnt encounter an error here after lots of computation has been done and wasted.
Also this should have a decent error explaining that it must be the same Python version as the target dist version, and that it must be the first in the PATH.

I would rather not have a config variable for this, or use the Windows registry's list of Pythons, as I feel that is extra code without any significant benefit -- I am assuming that anyone using PyOxidizer on Windows knows how to fiddle with their `PATH`.

If we were to support specifying the python.exe in the config, I would prefer that a custom `PATH` is put into the `extra_envs` of the packaging rule, and a global `extra_envs` is added if someone wants to avoid repeating a custom `PATH` in many rules.  The reason for this is that 1) a custom `PATH` is also often needed on unix for non-python build deps, and 2) the need for a custom python.exe on Windows should be a temporary thing as this is fixable in the standalone python image , and that is a pretty high priority to avoid lots of other hacks being needed for Windows only.

Reply:
I *really* do not want PyOxidizer using a system-provided Python. The `distutils` in the standalone Python distributions is pristine and can safely be used as long as we're not modifying it. We should be using it.

Reply:
I really do not either.  The problem is https://github.com/indygreg/python-build-standalone/issues/22 .  That needs to be solved, otherwise Windows is not supported because the python.exe provided can not run build dependencies which use built extensions, which is basically any non-trivial use-case.